xc/device.ml

```diff
@@ -1143,18 +1143,67 @@ module PCI = struct
     (* Sort into the order the devices were plugged *)
     List.sort (fun a b -> compare (fst a) (fst b)) pairs
 
-  let add ~xs pcidevs domid =
+  let _pci_add ~xc ~xs pcidev domid slot =
+    let sysfs_pci_dev = "/sys/bus/pci/devices/" in
+    let bus = 0 in
+    let func = 0 in
+    let pci_bdf = Printf.sprintf "%04x:%02x:%02x.%01x" domid bus slot func in
+    let page_size = 4096 in
+    if (Qemu.is_running ~xs domid) then
```

Comment:
else?

Reply:
Oops!
I think from the C xl code it would raise an `ERROR_FAIL` if the domain wasn't running, so I'll throw an exception